src/foam/core/Listener.js

```diff
@@ -113,6 +118,17 @@ foam.CLASS({
         configurable: true,
         enumerable: false
       });
+
+      if ( on.length > 0 ) {
```

Comment:
Add comment // Check avoids creating listener if 'on' is empty.

But maybe a simple approach is to get rid of the 'if' and replace this code with:

for ( var i = 0 ; i < on.length ; i++ ) {
          let o        = on[i].split('.');
          let objectOn = o.shift();
          let topic    = o;
          console.log(listener, topic);
          proto.onDetach(proto.sub.apply(proto, topic.concat(proto[this.name])));
        }